partiql-types/src/main/kotlin/org/partiql/types/StaticType.kt

```diff
@@ -479,17 +479,42 @@ data class StructType(
     val constraints: Set<TupleConstraint> = setOf(),
     override val metas: Map<String, Any> = mapOf(),
 ) : SingleType() {
+
+    public constructor(
+        fields: Map<String, StaticType>,
+        contentClosed: Boolean = false,
+        primaryKeyFields: List<String> = listOf(),
+        constraints: Set<TupleConstraint> = setOf(),
+        metas: Map<String, Any> = mapOf(),
+    ) : this(
+        fields.map { Field(it.key, it.value) },
+        contentClosed,
+        primaryKeyFields,
```

Comment:
Why is declaring `primaryKeyFields` and `contentClosed` limited to field being a Map?

Reply:
I think it's a good decision to contain `primaryKeyFields` and `contentClosed` to the previous public interface mainly b/c I'm of the opinion that they don't belong in `StructType` at all and this is a technical debt.